src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,92 +48,1243 @@
         super(account);
         startIndex = -1;
     }
+    
+    public void initForViewByQuestion() {
+        final ViewType viewType = ViewType.QUESTION;
+        
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionToResponseMap = bundle.getQuestionResponseMap();
+        questionPanels = new ArrayList<InstructorResultsQuestionTable>();
+        
+        for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> entry : questionToResponseMap.entrySet()) {
+            FeedbackQuestionAttributes question = entry.getKey();
+            List<FeedbackResponseAttributes> responses = entry.getValue();
+            
+            questionPanels.add(buildQuestionTableAndResponseRows(question, responses, viewType, ""));
+        }
+        
+    }
+    
+    /**
+     * Creates {@code InstructorFeedbackResultsSectionPanel}s for sectionPanels.
+     * 
+     * Iterates through the responses and creates panels and questions for them. Keeps track 
+     * of missing sections, teams and participants who do not have responses 
+     * and create panels for these missing sections, teams and participants.
+     * 
+     * TODO: simplify the logic in this method
+     */
+    public void initForViewByGiverQuestionRecipient() {
+  
+        if (!bundle.isComplete) {
+            // results page to be loaded by ajax instead 
+            buildSectionPanelsForForAjaxLoading(sections);
+            return;
+        }
+        
+        if (bundle.responses.isEmpty()) {
+            // no responses, nothing to initialize
+            return;
+        }
+        
+        setShouldCollapsed(bundle.responses.size() > 500);
+        
+        Map<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> sortedResponses 
+                     = bundle.getResponsesSortedByGiverQuestionRecipient(true);
+        
+        buildResponsesPanelsForGiverQuestionRecipient(sortedResponses);
+    }
+    
+    /**
+     * Creates {@code InstructorFeedbackResultsSectionPanel}s for sectionPanels.
+     * 
+     * Iterates through the responses and creates panels and questions for them. Keeps track 
+     * of missing sections, teams and participants who do not have responses 
+     * and create panels for these missing sections, teams and participants.
+     * 
+     * TODO: simplify the logic in this method
+     */
+    public void initForViewByRecipientQuestionGiver() {
 
-    /* 
-     * The next three methods are not covered in action test, but covered in UI tests.
+        if (!bundle.isComplete) {
+            // results page to be loaded by ajax instead 
+            buildSectionPanelsForForAjaxLoading(sections);
+            return;
+        }
+        
+        if (bundle.responses.isEmpty()) {
+            // no responses, nothing to initialize
+            return;
+        }
+        
+        setShouldCollapsed(bundle.responses.size() > 500);
```

Comment:
If these are going to be the same for all of the views or something, extract them into another method.